lib/bundle/stoplight-defaults.js

```diff
@@ -1,63 +1,137 @@
 "use strict";
 
-const { basename, join, stripRoot, dirname, extname } = require("@stoplight/path");
+const { join, stripRoot } = require("@stoplight/path");
 const {
+  getGenericDefaults,
   getDefaultsForOldJsonSchema,
   getDefaultsForOAS2,
   getDefaultsForOAS3,
 } = require("./defaults");
-const url = require("../util/url");
+const { parse, isHttp, isFileSystemPath } = require("../util/url");
+const KeyGenerator = require("./util/key-generator");
 
-function includeMidInFilePath (filepath, mid) {
-  if (!mid) {
-    return filepath;
+function StoplightKeyGenerator (root, { cwd, endpointUrl, srn }) {
+  KeyGenerator.call(this, root);
+  this._cwd = cwd;
+  this._endpointUrl = endpointUrl;
+  this._srn = srn;
+}
+
+StoplightKeyGenerator.prototype = Object.create(KeyGenerator.prototype, {
+  constructor: {
+    configurable: true,
+    writable: true,
+    value: StoplightKeyGenerator,
   }
+});
 
-  let ext = extname(filepath);
-  return join(dirname(filepath), basename(filepath, ext) + `.mid${mid}${ext || ".mid"}`);
-}
+StoplightKeyGenerator.prototype.processPath = function (path) {
```

Comment:
Curious - why the switch to tacking onto prototype?

Reply:
It's easier to override a certain method.
Previously, when we returned an object, it was more difficult, as certain variables weren't accessible outside.
I 
I could possibly use a plain `class`, but the library uses prototypes everywhere, so I stuck to it, since my hope is to back-port these changes to the upstream. One day.

Reply:
Ahh ok makes sense, in particular the consistency with upstream lib 👍.